### PR TITLE
Configure ruff as linter and formatter.

### PR DESCRIPTION
**Pull request by xbarra**
Co-authored-by: Arfima Dev <dev@arfima.com>
Signed-off-by: Xavier Barrachina Civera <xbarrachina@arfima.com>

### CODE REVIEW

**Comment by xbarra** [comment]
Following the discussion in #111 I configured ruff as linter and formatter, removing the others.

I am commenting the markdown pre-commit until #94 is fixed

File: pyproject.toml [hunk 1]
```diff
@@ -108,11 +106,8 @@ release = "scripts/release-versioning.sh"
 test = "pytest"
 docs = { shell = "cd docs && mkdocs serve", help = "Start the dev server for doc preview" }
 lint = "pre-commit run --all-files"
-isort = { cmd = "isort --check ." }
-black = { cmd = "black --check ." }
-flake8 = { cmd = "flake8 src" }
 mypy = { cmd = "mypy --install-types --non-interactive src" }
-all = {composite = ["pytest", "mypy", "isort", "black", "flake8"]}
+all = {composite = ["pytest", "mypy"]}
```
Comment (ciaransweet) [inline pyproject.toml:110]: Should all not also do the lint?

Comment (xbarra) [reply pyproject.toml]: you are right

**Comment by ciaransweet** [comment]
Out of interest @xbarra is this PR rebased onto my commit? I'm surprised to see so many formatting changes, especially as my PR did a lot of it, I'd only expect a few files from the black/ruff differences

**Comment by xbarra** [comment]
> Out of interest @xbarra is this PR rebased onto my commit? I'm surprised to see so many formatting changes, especially as my PR did a lot of it, I'd only expect a few files from the black/ruff differences

@ciaransweet 
It is rebased. The formatting changes are because the configuration in pyproject.toml adds several rules.

**Comment by ciaransweet** [comment]
> > Out of interest @xbarra is this PR rebased onto my commit? I'm surprised to see so many formatting changes, especially as my PR did a lot of it, I'd only expect a few files from the black/ruff differences
> 
> @ciaransweet It is rebased. The formatting changes are because the configuration in pyproject.toml adds several rules.

Gotcha, thanks!

**Pull request merged**